### PR TITLE
Implement a bot.placeEntity() function

### PR DESCRIPTION
**Pull request by u9g**

### CODE REVIEW

**Review by rom1504 — CHANGES_REQUESTED** [review]
could you add a test?

**Comment by u9g** [comment]
wip still

**Comment by rom1504** [comment]
I like it but it's old and stale, reopen if you want to work on it

**Comment by u9g** [comment]
awaiting https://github.com/PrismarineJS/prismarine-item/pull/33 to be merged, then needs to be replaced with having the function here

File: docs/api.md [hunk 1]
```diff
@@ -1656,6 +1656,17 @@ This function also returns a `Promise`, with `void` as its argument upon complet
 
 The new block will be placed at `referenceBlock.position.plus(faceVector)`.
 
+#### bot.placeEntity(referenceBlock, faceVector, cb)
```
Comment (rom1504) [inline docs/api.md:1659]: let's remove the cb (not adding it for new methods)

File: lib/plugins/place_entity.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    assert(['end_crystal', 'boat', 'spawn_egg', 'armor_stand'].includes(type), 'Unimplemented')
+
+    let name = bot.heldItem.name // used for finding entity after spawn
+      .replace(/.+_boat/, 'boat')
+
+    if (name.endsWith('spawn_egg')) {
+      name = bot.heldItem.spawnEggMobName
+    }
+    // Look at the center of the face
+    let dx = 0.5 + faceVector.x * 0.5
```
Comment (rom1504) [inline lib/plugins/place_entity.js:24]: is there any code from all this you could put in common with place block ? looks like yes

File: lib/plugins/place_entity.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+      bot.on('entitySpawn', listener)
+    })
+  }
+
+  bot.placeEntity = callbackify(placeEntity)
+  bot._placeEntityWIthOptions = placeEntityWithOptions
+}
+
+function vectorToDirection (v) {
```
Comment (rom1504) [inline lib/plugins/place_entity.js:160]: don't we have this function elsewhere already ?

**Comment by rom1504** [comment]
overall lgtm, let's just try to remove some code duplication

File: lib/plugins/place_entity.js (hunk 2 above)
Comment (u9g) [reply lib/plugins/place_entity.js:24]: most of the place logic is the same, yeah

File: lib/plugins/place_entity.js (hunk 3 above)
Comment (u9g) [reply lib/plugins/place_entity.js:160]: yeah, in placeBlock :p maybe should go in util?

Comment (rom1504) [reply lib/plugins/place_entity.js:160]: yeah something like that

File: lib/plugins/place_entity.js (hunk 2 above)
Comment (rom1504) [reply lib/plugins/place_entity.js:24]: so try to do a function with the common stuff and call it here and in place block
will decrease the maintenance cost

File: docs/api.md (hunk 1 above)
Comment (u9g) [reply docs/api.md]: Done

File: lib/plugins/place_entity.js (hunk 2 above)
Comment (u9g) [reply lib/plugins/place_entity.js]: Done

File: lib/plugins/place_entity.js (hunk 3 above)
Comment (u9g) [reply lib/plugins/place_entity.js]: Done

**Review by rom1504 — APPROVED** [review]

**Pull request merged**